src/plots/cartesian/index.js

```diff
@@ -515,6 +524,11 @@ function purgeSubplotLayers(layers, fullLayout) {
     }
 }
 
+function removeSubplotExtras(subplotId, fullLayout) {
+    fullLayout._draggers.selectAll('g.' + subplotId).remove();
+    fullLayout._defs.select('#clip' + fullLayout._uid + subplotId + 'plot').remove();
```

Comment:
What about axis titles? They are removed [above](https://github.com/plotly/plotly.js/blob/315272be116c29c108487537dcf6c0b2f7e7a2c2/src/plots/cartesian/index.js#L296) in an old vs new axis loop, could we combine that in `removeSubplotExtras`?

Reply:
Titles are per axis, because there's only ever one copy per axis, so they're inserted with no reference to the subplot and it would be incorrect to tie their removal to the disappearance of any given subplot.